tools/bowtie2.wdl

```diff
@@ -0,0 +1,447 @@
+        }
+        n_ceil: {
+            description: "func for maximum number of non-A/C/G/Ts permitted in alignment",
+            help: "Sets a function governing the maximum number of ambiguous characters (usually Ns and/or .s) allowed in a read as a function of read length. For instance, specifying -L,0,0.15 sets the N-ceiling function f to f(x) = 0 + 0.15 * x, where x is the read length. Reads exceeding this ceiling are filtered out."
+        }
+    }
+
+    input {
+        File bowtie_db_tar_gz
+        File read_one_fastq_gz
+        ReadGroup read_group
+        String prefix
+        File? read_two_fastq_gz
+        String? addl_rg_text
+        String? sam_opt_config
+        Int? skip
+        Int? upto
+        Int? match_bonus
+        Int? max_aln_report
+        Bowtie2Function score_min = Bowtie2Function {
+            function_type: "L",
+            constant: -0.6,
+            coefficient: -0.6,
+        }
+        Bowtie2Function interval_seed_substrings = Bowtie2Function {
+            function_type: "S",
+            constant: 1,
+            coefficient: 1.15,
+        }
+        Bowtie2Function n_ceil = Bowtie2Function {
+            function_type: "L",
+            constant: 0,
+            coefficient: 0.15,
+        }
+        Pair[Int, Int] read_gap_open_extend = (5, 3)
+        Pair[Int, Int] ref_gap_open_extend = (5, 3)
+        Pair[Int, Int] mismatch_penalty = (6, 2)
+        Boolean end_to_end = true  # false: --local
+        Boolean non_deterministic = false
+        Boolean ignore_quals = false
+        Boolean nofw = false
+        Boolean norc = false
+        Boolean no_1mm_upfront = false
+        Boolean report_all_alignments = false
+        Boolean no_mixed = false
+        Boolean no_discordant = false
+        Boolean dovetail = false
+        Boolean no_contain = false
+        Boolean no_overlap = false
+        Boolean write_unpaired_unaligned = false
+        Boolean write_unpaired_aligned = false
+        Boolean write_paired_discordant = false
+        Boolean write_paired_concordant = false
+        Boolean metrics_file = false
+        Boolean metrics_stderr = false
+        Boolean no_unal = false
+        Boolean omit_sec_seq = false
+        Boolean sam_no_qname_trunc = false
+        Boolean xeq = false
+        Boolean soft_clipped_unmapped_tlen = false
+        Boolean sam_append_comment = false
+        Boolean reorder = false
+        Boolean qc_filter = false
+        Int trim5 = 0
+        Int trim3 = 0
+        Int seed_mismatch = 0
+        Int seed_substring = 22
+        Int dpad = 15
+        Int gbar = 4
+        Int non_actg_penalty = 1
+        Int min_fragment_len = 0
+        Int max_fragment_len = 500
+        Int metrics_interval = 1
+        Int ncpu = 1
```

Comment:
Is 1 thread enough for reasonable performance?

Reply:
1 is the default in `bowtie2`. The "optimal" point is probably 8 (https://www.jefftk.com/p/benchmarking-bowtie2-threading)